Edited user model, added registration and log in forms

mayasa3 opened the pull request:
- Renamed login app as accounts to prevent naming issues and updated dependencies

- Renamed variables

- Updated UserAccount model to extend base User model

- Updated models in searchJob and user_notification apps to fix an issue with foreign keys referencing users

- Forms, views and URLs for registration, log in and log out features

- HTML pages for registration and login


@bardielle @benipeled @guy9050

bardielle reviewed (changes requested):
Split it into multiple commits
it cannot be read like that...

mayasa3 committed:
Updated user model and renamed app

- UserAccount model now extends base user

- 'login' app renamed as 'accounts' to prevent naming issues

- Dependencies updated to match

- JobAlert and UserNotification models updated to call user

mayasa3 committed:
Added forms and views for registration, log in and log out

- Views updated for registration, log in and log out

- Added forms and URLs for registration and log in

- HTML pages for the registration and log in forms

mayasa3 commented:
I split it into 2 commits, is it ok like this? @bardielle

bardielle commented on accounts/admin.py:
```diff
@@ -0,0 +1,38 @@
+)
+
+admin.site.unregister(User)
+admin.site.register(User, UserAdmin)
```
@ifireball is it fine to unregister user like that? Why do we need it?

mayasa3 replied:
A lot of the instructions I found for adding fields to the admin panel said to unregister and re-register it, but I can remove that part

bardielle replied:
Please fix it

bardielle commented on searchJob/models.py:
```diff
@@ -0,0 +1,64 @@
+from django.db import models
+
+
+class Company_type(models.TextChoices):
```
class names are with capital letters... Company_type -> CompanyType

bardielle commented on searchJob/models.py:
```diff
@@ -0,0 +1,64 @@
+    UNSPECIFIED = 'UNSPECIFIED', 'UNSPECIFIED'
+
+
+class Job_scope(models.TextChoices):
```
Job_scope ->JobScope

bardielle commented on searchJob/models.py:
```diff
@@ -0,0 +1,64 @@
+    UNSPECIFIED = 'UNSPECIFIED', 'UNSPECIFIED'
+
+
+class Application_status(models.TextChoices):
```
Application_status ->ApplicationStatus

bardielle commented on searchJob/models.py:
```diff
@@ -0,0 +1,64 @@
+
+
+class UserApplication(models.Model):
+    UserAccount_id = models.ForeignKey("accounts.UserAccount", on_delete=models.CASCADE, blank=True, null=True)
```
UserAccount_id ->user_account_id

bardielle replied:
??

bardielle commented on searchJob/models.py:
```diff
@@ -0,0 +1,64 @@
+
+class UserApplication(models.Model):
+    UserAccount_id = models.ForeignKey("accounts.UserAccount", on_delete=models.CASCADE, blank=True, null=True)
+    jobPost_id = models.ForeignKey(JobPost, on_delete=models.CASCADE, blank=True, null=True)
```
job_post_id

bardielle replied:
??

bardielle commented on accounts/urls.py:
```diff
@@ -0,0 +1,24 @@
+
+The `urlpatterns` list routes URLs to views. For more information please see:
+    https://docs.djangoproject.com/en/3.1/topics/http/urls/
+Examples:
```
remove un-relevant comments....

bardielle replied:
??

bardielle commented on accounts/views.py:
```diff
@@ -1,3 +1,44 @@
+        if user_form.is_valid() and account_form.is_valid():
+            user = user_form.save()
+            user.save()
+            useraccount = account_form.save(commit=False)
```
useraccount - >user_account

bardielle replied:
@mayasa3

bardielle replied:
@mayasa3 this comment is still relevant

mayasa3 commented:
Renamed everything and changed the admin file

bardielle commented on accounts/admin.py:
```diff
@@ -34,5 +14,3 @@ class UserAdmin(BaseUserAdmin):
     }),
 )
 
-admin.site.unregister(User)
```
fix my comments on the commit it was mentioned...

mayasa3 replied:
Does it look ok now? Had some trouble with the rebase

bardielle commented on searchJob/models.py:
```diff
@@ -1,7 +1,7 @@
 
 
-class Company_type(models.TextChoices):
+class CompanyType(models.TextChoices):
```
fix my comments on the commit it was mentioned...

bardielle commented on searchJob/models.py:
```diff
@@ -12,15 +12,15 @@ class Company_type(models.TextChoices):
 
 
-class Job_scope(models.TextChoices):
+class JobScope(models.TextChoices):
```
fix my comments on the commit it was mentioned...

bardielle commented on searchJob/models.py:
```diff
@@ -12,15 +12,15 @@ class Company_type(models.TextChoices):
 
 
-class Application_status(models.TextChoices):
+class ApplicationStatus(models.TextChoices):
```
fix my comments on the commit it was mentioned...

bardielle commented on searchJob/models.py:
```diff
@@ -55,8 +55,8 @@ def __str__(self):
 class UserApplication(models.Model):
-    UserAccount_id = models.ForeignKey("accounts.UserAccount", on_delete=models.CASCADE, blank=True, null=True)
-    jobPost_id = models.ForeignKey(JobPost, on_delete=models.CASCADE, blank=True, null=True)
+    user_account_id = models.ForeignKey("accounts.UserAccount", on_delete=models.CASCADE, blank=True, null=True)
```
fix my comments on the commit it was mentioned...

mayasa3 commented:
Updated the commits, is there anything else left to fix?

bardielle commented on accounts/admin.py:
```diff
@@ -0,0 +1,35 @@
+from accounts.models import UserAccount
+
+
+admin.site.register(UserAccount)
```
Move it to the end of the file where all the other registers

bardielle commented on accounts/admin.py:
```diff
@@ -0,0 +1,35 @@
+    }),
+)
+
+admin.site.unregister(User)
```
Why do you need "unregister" and again "register"?

bardielle commented on accounts/admin.py:
```diff
@@ -0,0 +1,35 @@
+admin.site.register(UserAccount)
+
+
+class AccountInline(admin.StackedInline):
```
Why all those classes in the admin.py file? they need to be in the models.py
@ifireball Am I wrong?

bardielle commented on accounts/migrations/0003_auto_20201213_1916.py:
```diff
@@ -0,0 +1,55 @@
+# Generated by Django 3.1.4 on 2020-12-13 19:16
+
```
You dont need multiple migration files as far as I know
@ifireball fix me if Im wrong....
